### DISCUSSION TITLE
How to get packages to stay flat?

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [pnpm/pnpm#4063] [Q&A]
Hey, I have a package that's just a `package.json` with a bunch of CLI apps that I use so I don't have like 50 different devDependencies. However, when I run `pnpm bin_name` it does not find it. And when I run `pnpx` it asks me if I want to download it even though it's already installed under my grouped package. Is there anyway I can address this without having to --shamefully-hoist (why name it that --- ugh)?

**Comment by zkochan** [comment]
Sounds like related to this discussion: https://github.com/pnpm/pnpm/discussions/3958

Instead of `--shamefull-hoist`, you may use `publish-hoist-pattern` to specify which packages you want to hoist. Like

.npmrc:
```
public-hoist-pattern[]=foo
public-hoist-pattern[]=bar
public-hoist-pattern[]=@company/plugin-*
```
